src/Control/Controller.php

```diff
@@ -680,11 +680,11 @@ public static function join_links($arg = null)
         foreach ($args as $arg) {
             // Find fragment identifier - keep the last one
             if (strpos($arg, '#') !== false) {
-                list($arg, $fragmentIdentifier) = explode('#', $arg, 2);
+                list($arg, $fragmentIdentifier) = explode('#', (string) $arg, 2);
```

Comment:
Probably needs `$arg = (string) $arg;` as the first line in the loop and then it's all good for all the checks and usages.